src/main/java/de/tum/in/www1/artemis/repository/CourseCodeOfConductAgreementRepository.java

```diff
@@ -0,0 +1,31 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.CourseCodeOfConductAgreement;
+
+/**
+ * Spring Data repository for the Code of Conduct Agreement entity.
+ */
+@Repository
+public interface CourseCodeOfConductAgreementRepository extends JpaRepository<CourseCodeOfConductAgreement, Long> {
+
+    /**
+     * Find the user's agreement to a course's code of conduct.
+     *
+     * @param courseId the ID of the code of conduct's course
+     * @param userId   the user's ID
+     * @return the user's agreement to the course's code of conduct
+     */
+    @Query("""
+            SELECT c
+            FROM CourseCodeOfConductAgreement c
+            WHERE c.course.id = :courseId AND c.user.id = :userId
+            """)
+    Optional<CourseCodeOfConductAgreement> findByCourseIdAndUserId(@Param("courseId") Long courseId, @Param("userId") Long userId);
```

Comment:
Optional: I think you can even leave the query out and let spa generate it itself using the method name.

Reply:
I didn't know, thanks! 45fbbc7319e6e8bb6df676f8b676671e698f4896